code/statistical_tests.py

```diff
@@ -10,20 +10,45 @@
 from lexicalrichness import LexicalRichness
 import textstat
 
-#splitting the dfs 
-data1 = pd.read_csv(r'C:\Users\rasik\Documents\Independent Study\data\translated data\1_15_final.csv')
-data2= pd.read_csv(r'C:\Users\rasik\Documents\Independent Study\data\translated data\16_35_final.csv')
-data3 = pd.read_csv(r'C:\Users\rasik\Documents\Independent Study\data\translated data\36_150_final.csv')
-data4 = pd.read_csv(r'C:\Users\rasik\Documents\Independent Study\data\translated data\151_500_final.csv')
-data5 = pd.read_csv(r'C:\Users\rasik\Documents\Independent Study\data\translated data\501_1500_final.csv')
-data6 = pd.read_csv(r'C:\Users\rasik\Documents\Independent Study\data\translated data\1501_5000_final.csv')
-data7 =pd.read_csv(r'C:\Users\rasik\Documents\Independent Study\data\translated data\5001_final.csv')
-
-pre_150 = pd.concat([data1, data2, data3])
-#print(len(pre_150))
-
-post_150 = pd.concat([data4, data5, data6, data7])
-#print(len(post_150))
+# #splitting the dfs 
+# data1 = pd.read_csv(r'C:\Users\rasik\Documents\Independent Study\data\translated data\1_15_final.csv')
+# data2= pd.read_csv(r'C:\Users\rasik\Documents\Independent Study\data\translated data\16_35_final.csv')
+# data3 = pd.read_csv(r'C:\Users\rasik\Documents\Independent Study\data\translated data\36_150_final.csv')
+# data4 = pd.read_csv(r'C:\Users\rasik\Documents\Independent Study\data\translated data\151_500_final.csv')
+# data5 = pd.read_csv(r'C:\Users\rasik\Documents\Independent Study\data\translated data\501_1500_final.csv')
+# data6 = pd.read_csv(r'C:\Users\rasik\Documents\Independent Study\data\translated data\1501_5000_final.csv')
+# data7 =pd.read_csv(r'C:\Users\rasik\Documents\Independent Study\data\translated data\5001_final.csv')
+
+# pre_150 = pd.concat([data1, data2, data3])
+# #print(len(pre_150))
+
+# post_150 = pd.concat([data4, data5, data6, data7])
+# #print(len(post_150))
+
+data = pd.read_csv('sampled_dataset.csv')
+
+# Define the first and second groups
+group1_categories = ['1 to 5', '6 to 15', '16 to 50']
+group2_categories = ['51 to 150', '151 to 500', '501 to 1500', '1501 to 5000', '5000+']
+
+# Split the dataset into two groups
+pre_150 = data[data['instance group'].isin(group1_categories)]
+post_150 = data[data['instance group'].isin(group2_categories)]
+
+
+# Calculate the confidence interval for Pearson's r
+def pearson_confidence_interval(r, n, confidence_level=0.95):
```

Comment:
sometimes, writing your own code for something well established can lead to errors. Is there stat library doing this? you can also write tests to make sure this routine is correct.